core/deployment/src/test/java/io/quarkus/deployment/pkg/steps/JarResultBuildStepTest.java

```diff
@@ -2,27 +2,63 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
+import java.io.File;
+import java.io.FileOutputStream;
+import java.math.BigInteger;
 import java.nio.file.Path;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.security.Security;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.util.Calendar;
+import java.util.Date;
 import java.util.Set;
 import java.util.jar.JarEntry;
 import java.util.jar.JarFile;
 import java.util.jar.Manifest;
+import java.util.zip.ZipFile;
 
+import org.bouncycastle.asn1.ASN1ObjectIdentifier;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x509.BasicConstraints;
+import org.bouncycastle.cert.CertIOException;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.operator.ContentSigner;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 
+import jdk.security.jarsigner.JarSigner;
+
 /**
  * Test for {@link JarResultBuildStep}
  */
 class JarResultBuildStepTest {
 
     @Test
-    void should_unsign_jar_when_filtered(@TempDir Path tempDir) throws Exception {
-        Path signedJarFilePath = Path.of(getClass().getClassLoader().getResource("signed.jar").toURI());
-        Path jarFilePath = tempDir.resolve("unsigned.jar");
-        JarResultBuildStep.filterJarFile(signedJarFilePath, jarFilePath,
-                Set.of("org/eclipse/jgit/transport/sshd/SshdSessionFactory.class"));
-        try (JarFile jarFile = new JarFile(jarFilePath.toFile())) {
+    void should_unsign_jar_when_filtered(@TempDir Path tmp) throws Exception {
```

Comment:
Can you keep the same parameter name? That will make it easier to notice what really changed here